hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -678,6 +694,19 @@ private ByteString readStateMachineData(
     }
   }
 
+  private void getReadStateMachineFuture(
+      CompletableFuture<ByteString> readStateMachineFuture,
+      ContainerCommandRequestProto requestProto, LogEntryProto entry) {
+    try {
+      readStateMachineFuture.complete(
+          readStateMachineData(requestProto, entry.getTerm(),
+              entry.getIndex()));
+    } catch (IOException e) {
+      metrics.incNumReadStateMachineFails();
+      readStateMachineFuture.completeExceptionally(e);
```

Comment:
we should mark the statemachine unhealthy and container unhealthy if we fail here ?

Reply:
stateMachine will be marked unhealthy right at readStateMachineData() call. Similarly, Container will be marked unhealthy in HddsDispatcher if readChunk fails.